enhancements/baremetal/ztp-metal3.md

```diff
@@ -0,0 +1,321 @@
+  accept additional Ignition configuration via a new annotation
+  `baremetal.openshift.io/ignition-override-uri`.
+
+- Additionally, ICC will recognize the label linking `BareMetalHost`s to
+  `InfraEnv` resources from the assisted service. When such a label is present,
+  ICC will wait for the non-empty Ignition override URI to be provided.
+  Otherwise, it will proceed with the image building.
+
+- [Ironic agent][ironic-agent] currently contains a custom deployment plugin
+  (*custom deploy method* in baremetal-operator terms, *custom deploy step* in
+  Ironic terms) that invokes `coreos-installer` for the image installation.
+  A new custom deploy method `install_assisted` will be added to start the
+  assisted agent at the right moment.
+
+- The assisted service (more specifically, BMAC) will stop explicitly disabling
+  *inspection* on `BareMetalHost`s it manages, will start using a different
+  process based on the  new `install_assisted` deploy method and stop rebooting
+  the machine in the end of the process.
+
+- BMAC will also set the Ignition URI with the assisted agent configuration on
+  any `PreprovisioningImage` objects linked to `BareMetalHost` objects it
+  manages.
+
+### User Stories
+
+This change itself does not add new user stories, but rather enables existing
+features to work for ZTP:
+
+- As an operator, I want to ensure that all OpenShift nodes are deployed with
+  known firmware settings.
+
+- As an operator of hardware that does not support virtual media, I want to
+  deploy OpenShift nodes using iPXE network boot.
+
+- As an operator, I want OpenShift to ensure that the deployment ISO is
+  correctly disconnected on installation finish and cannot be accidentally
+  booted into.
+
+### API Extensions
+
+All required API extensions are already present.
+
+A new annotation will be recognized on `PreprovisioningImage` resources:
+`baremetal.openshift.io/ignition-override-uri`.
+
+### Implementation Details/Notes/Constraints
+
+#### Coordination between agents
+
+The first agent to start will be the Ironic agent since it is responsible
+for coordinating the whole flow and (indirectly through Ironic) updating the
+`BareMetalHost`'s status. *Inspection* and additional configuration (RAID etc)
+will be handled by the Ironic agent alone.
+
+BMAC will be able to stop synchronizing the inventory to `BareMetalHost`
+objects since this task will be handled by Ironic and baremetal-operator.
+
+Currently, the provisioning is initiated by BMAC (the assisted component
+responsible for interactions with baremetal-operator) by applying the following
+change to the `BareMetalHost`'s spec:
+
+```yaml
+Image:
+  URL: url/of/assisted/agent.iso
+  DiskFormat: live-iso
+```
+
+After this enhancement, BMAC will instead apply:
+
+```yaml
+CustomDeploy:
+  Method: install_assisted
+```
+
+This instructs Ironic to find and execute a *custom deploy step* called
+`install_assisted`, which will be provided by an OpenShift specific Ironic
```

Comment:
I don't think this step should be called `install_assisted` if all it does is start the assisted agent and not actually begin an install. Something like `start_assisted` might be more appropriate.